backend_new/test/integration/unit-type.e2e-spec.ts

```diff
@@ -0,0 +1,140 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { INestApplication } from '@nestjs/common';
+import * as request from 'supertest';
+import { AppModule } from '../../src/app.module';
+import { PrismaService } from '../../src/services/prisma.service';
+import { unitTypeFactory } from '../../prisma/seed-helpers/unit-type-factory';
+import { UnitTypeCreate } from '../../src/dtos/unit-types/unit-type-create.dto';
+import { UnitTypeUpdate } from '../../src/dtos/unit-types/unit-type-update.dto';
+import { IdDTO } from 'src/dtos/shared/id.dto';
+import { randomUUID } from 'crypto';
+
+describe('UnitType Controller Tests', () => {
+  let app: INestApplication;
+  let prisma: PrismaService;
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    prisma = moduleFixture.get<PrismaService>(PrismaService);
+    await app.init();
+  });
+
+  it('testing list endpoint', async () => {
+    const unitTypeA = await prisma.unitTypes.create({
+      data: unitTypeFactory(7),
+    });
+    const unitTypeB = await prisma.unitTypes.create({
+      data: unitTypeFactory(8),
+    });
+
+    const res = await request(app.getHttpServer())
+      .get(`/unitTypes?`)
+      .expect(200);
+
+    expect(res.body.length).toEqual(2);
+    const sortedResults = res.body.sort(
+      (a, b) => a.numBedrooms - b.numBedrooms,
+    );
+    expect(sortedResults[0].name).toEqual(unitTypeA.name);
+    expect(sortedResults[1].name).toEqual(unitTypeB.name);
+  });
```

Comment:
I'm not a big fan of us relying on this test going first. For example if you move the "testing retrieve endpoint" test above this one it fails. If we want to keep this pattern I'd recommend changing the expect statements to be a little more directed. Such as:
```
    expect(res.body.length).toBeGreaterThanOrEqual(2);
    const unitTypeNames = res.body.map((value) => value.name);
    expect(unitTypeNames).toContain(unitTypeA.name);
    expect(unitTypeNames).toContain(unitTypeB.name);
```

Reply:
nice call out! got this updated, lemme know what you think